relay-server/src/actors/project_buffer.rs

```diff
@@ -263,10 +270,51 @@ impl BufferService {
             )))
     }
 
+    /// Saves the provided buffer to the disk.
+    ///
+    /// It returns an error if the spooling failed.
+    async fn do_spool(
+        connection: &mut PoolConnection<Sqlite>,
+        buffer: BTreeMap<QueueKey, Vec<ManagedEnvelope>>,
+    ) -> Result<(), BufferError> {
+        // A builder type for constructing queries at runtime.
+        // This by default creates a prepared sql statement, which is cached and
+        // re-used for sequential queries.
+        let mut query_builder: QueryBuilder<Sqlite> =
+            QueryBuilder::new("INSERT INTO envelopes (own_key, sampling_key, envelope) ");
+
+        // Flatten all the envelopes
+        let envelopes = buffer.into_iter().flat_map(|(k, vals)| {
+            vals.into_iter()
+                .map(move |v| (k, v.into_envelope().to_vec()))
```

Comment:
This loop can take quite a while, and we're not yielding. This means that the thread on the main runtime that runs this future is going to be blocked for a long time. Should we either yield here, or defer this into a `spawn_blocking`? The latter is semantically wrong, and the CPU time has to be spent, but it should be spent in a context where we do not block other futures for too long.

Example:

```rust
let mut envelopes = Vec::new();
for k, vals in buffer {
  for val in vals {
    envelopes.insert((k, val.into_envelope().to_vec()));
    tokio::task::yield_now().await;  // TODO: consider doing this every Nth iteration
  }
}
```